topo_processor/metadata/collection_factory.py

```diff
@@ -1,28 +1,37 @@
+import asyncio
 import os
 
 from linz_logger import get_log
 
-from topo_processor.metadata.loaders import repo
+from topo_processor.metadata.loaders import loader_repo
+from topo_processor.metadata.validators import validator_repo
 from topo_processor.util.time import time_in_ms
 
 from .collection import Collection
 from .data_type import DataType
 from .item import Item
 
 
-def create_collection(path: str, data_type: DataType) -> Collection:
+async def create_collection(path: str, data_type: DataType) -> Collection:
     start_time = time_in_ms()
     collection = Collection("title", "description", "license", data_type)
-    create_item(collection, path)
-    get_log().debug("CreateCollection", title=collection.title, data_type=data_type, duration=time_in_ms() - start_time)
+    await create_items(collection, path)
+    get_log().debug("Collection Created", title=collection.title, data_type=data_type, duration=time_in_ms() - start_time)
     return collection
 
 
-def create_item(collection: Collection, path: str) -> None:
+async def process_item(item):
+    await loader_repo.add_metadata(item)
+    await validator_repo.check_validity(item)
+
+
+async def create_items(collection: Collection, path: str) -> None:
     start_time = time_in_ms()
+    items_to_process = []
     for f in os.listdir(path):
         file_path = os.path.join(path, f)
         item = Item(file_path, collection)
-        repo.add_metadata(item)
+        items_to_process.append(process_item(item))
         collection.items.append(item)
-    get_log().debug("CreateItem", path=path, duration=time_in_ms() - start_time)
+    await asyncio.gather(*items_to_process)
+    get_log().debug(f"{len(collection.items)} Items Created", path=path, duration=time_in_ms() - start_time)
```

Comment:
Generally avoid putting numbers into strings for logs, this would be better as a `count=len(collection.items)`